plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -392,47 +397,30 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
   }
 
   async facets(request: EntityFacetsRequest): Promise<EntityFacetsResponse> {
-    const { entities } = await this.entities({
-      filter: request.filter,
-      authorizationToken: request.authorizationToken,
-    });
-
     const facets: EntityFacetsResponse['facets'] = {};
+    const db = this.database;
 
     for (const facet of request.facets) {
-      const values = entities
-        .map(entity => {
-          // TODO(freben): Generalize this code to handle any field that may
-          // have dots in its key?
-          if (facet.startsWith('metadata.annotations.')) {
-            return entity.metadata.annotations?.[
-              facet.substring('metadata.annotations.'.length)
-            ];
-          } else if (facet.startsWith('metadata.labels.')) {
-            return entity.metadata.labels?.[
-              facet.substring('metadata.labels.'.length)
-            ];
-          }
-          return lodash.get(entity, facet);
-        })
-        .flatMap(field => {
-          if (typeof field === 'string') {
-            return [field];
-          } else if (Array.isArray(field)) {
-            return field.filter(i => typeof i === 'string');
-          }
-          return [];
+      const dbQuery = db<DbSearchRow>('search')
+        .join('final_entities', 'search.entity_id', 'final_entities.entity_id')
+        .where('search.key', facet.toLowerCase())
+        .count('search.entity_id as count')
+        .select({
+          value: 'search.original_value',
         })
-        .sort();
+        .groupBy('search.original_value');
 
-      const counts = lodash.countBy(values, lodash.identity);
+      if (request?.filter) {
+        parseFilter(request.filter, dbQuery, db, false, 'search.entity_id');
+      }
+
+      const result = await dbQuery;
 
-      facets[facet] = Object.entries(counts).map(([value, count]) => ({
-        value,
-        count,
+      facets[facet] = result.map(data => ({
+        value: data.value as string,
+        count: data.count as number,
```

Comment:
Seems you need to do `Number(data.count)` instead, if knex returns the thing as a string.

Reply:
(i pushed a fix for this, let's see if that makes the build pass - remember to pull from remote if you continue development)